src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +106,56 @@ public Object instantiate() throws ConstructionException {
+            LogContext mdcContextInjectionPointFieldValue;
             if (d.getInjectionPoint() instanceof FieldInjectionPoint) {
-                FieldInjectionPoint fd = (FieldInjectionPoint) d.getInjectionPoint();
-                Object value = checkNull(fd, providers.get(d).instantiate());
-                Field field = fd.getMember();
-
+                mdcContextInjectionPointFieldValue = LogContext.create();
+                fd = (FieldInjectionPoint) d.getInjectionPoint();
+                value = checkNull(fd, providers.get(d).instantiate());
+                field = fd.getMember();
                 try {
                     logger.trace("Setting field {} with arguments {}", field, value);
                     field.setAccessible(true);
                     field.set(instance, value);
+                    mdcContextInjectionPointFieldValue.put("org.grouplens.graph.injectionPoint.field", fd.toString());
+                    mdcContextInjectionPointFieldValue.put("org.grouplens.grapht.injectionPoint.field.member", field.toString());
+                    mdcContextInjectionPointFieldValue.put("org.grouplens.grapht.injectionPoint.field.value", value.toString());
                 } catch (IllegalAccessException e) {
                     throw new ConstructionException(fd, e);
                 }
+                  finally {
+                    mdcContextInjectionPointFieldValue.finish();
+                }
             } else if (d.getInjectionPoint() instanceof SetterInjectionPoint) {
                 // collect parameters before invoking
+                LogContext mdcContextSetterInjectionPoint = LogContext.create();
                 SetterInjectionPoint sd = (SetterInjectionPoint) d.getInjectionPoint();
                 InjectionArgs args = settersAndArguments.get(sd.getMember());
                 Method setter = sd.getMember();
-
                 if (args == null) {
                     // first encounter of this method
                     args = new InjectionArgs(setter.getParameterTypes().length);
                     settersAndArguments.put(setter, args);
                 }
-                
+                logger.trace("Injection point in progress with arguments {}", args);
+                logger.trace("Injection point in progress with method {}", setter);
                 Instantiator provider = providers.get(d);
+                mdcContextSetterInjectionPoint.put("org.grouplens.grapht.currentProvide", provider.toString());
+                mdcContextSetterInjectionPoint.put("org.grouplens.grapht.injectionPoint.setterinjectionpoint", sd.toString());
+                mdcContextSetterInjectionPoint.put("org.grouplens.grapht.injectionPoint.setterinjectionpoint,method", setter.toString());
+                mdcContextSetterInjectionPoint.put("org.grouplens.grapht.injectionPoint.setterinjectionpoint.args", args.toString());
                 args.set(sd.getParameterIndex(), checkNull(sd, provider.instantiate()));
```

Comment:
Great, we have the right ordering now; we just need a `try` for this log context.